app/code/Magento/Catalog/Helper/Product/ProductList.php

```diff
@@ -41,9 +42,11 @@ class ProductList
      * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
      */
     public function __construct(
-        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\Framework\Registry $coreRegistry
```

Comment:
I think this dependancy should be injected with a default of null. Then when setting to the `$this->coreRegistry` you can check for null and load from the object manager. See http://devdocs.magento.com/guides/v2.1/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter for a bit more details.